core/src/main/scala/kafka/controller/ReplicaStateMachine.scala

```diff
@@ -26,22 +26,23 @@ import kafka.zk.KafkaZkClient.UpdateLeaderAndIsrResult
 import kafka.zk.TopicPartitionStateZNode
 import org.apache.kafka.common.TopicPartition
 import org.apache.kafka.common.errors.ControllerMovedException
+import org.apache.kafka.common.utils.Time
 import org.apache.zookeeper.KeeperException.Code
 import scala.collection.{Seq, mutable}
 
 abstract class ReplicaStateMachine(controllerContext: ControllerContext) extends Logging {
   /**
    * Invoked on successful controller election.
    */
-  def startup(): Unit = {
-    info("Initializing replica state")
+  def startup(controllerStartMs: Long = Time.SYSTEM.milliseconds): Unit = {
+    info(s"Initializing replica state ${KafkaController.timing(-1, controllerStartMs)}")
     initializeReplicaState()
-    info("Triggering online replica state changes")
     val (onlineReplicas, offlineReplicas) = controllerContext.onlineAndOfflineReplicas
+    info(s"Triggering online replica state changes for ${onlineReplicas.size} replicas ${KafkaController.timing(-1, controllerStartMs)}")
     handleStateChanges(onlineReplicas.toSeq, OnlineReplica)
-    info("Triggering offline replica state changes")
+    info(s"Triggering offline replica state changes for ${offlineReplicas.size} replicas ${KafkaController.timing(-1, controllerStartMs)}")
     handleStateChanges(offlineReplicas.toSeq, OfflineReplica)
-    debug(s"Started replica state machine with initial state -> ${controllerContext.replicaStates}")
+    debug(s"Started replica state machine with initial state -> ${controllerContext.replicaStates} ${KafkaController.timing(-1, controllerStartMs)}")
```

Comment:
Suggestion: since the replicaStates could yield a very long string, I'd suggest moving the 
KafkaController.timing(..)
before it.

Reply:
You're right, it is a very long string. :-)  I'll do that.